private-templates-service/client/src/components/Common/SpinnerModal/SpinnerModal.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { CenteredSpinner } from './styled';
+import { connect } from 'react-redux';
+import { RootState } from '../../../store/rootReducer';
+import { closeModal } from '../../../store/page/actions';
+import { BackDrop } from '../ShareModal/styled';
+import { SpinnerSize } from 'office-ui-fabric-react';
+
+const mapStateToProps = (state: RootState) => {
+  return {
+    isFetching: state.currentTemplate.isFetching
+  }
+}
+
+const mapDispatchToProps = (dispatch: any) => {
+  return {
+    closeModal: () => {
+      dispatch(closeModal());
+    }
+  }
+}
+
+interface Props {
+  isFetching: boolean;
+  closeModal: () => void;
+  closeAction?: () => void;
+}
+
+class SpinnerModal extends React.Component<Props>{
+  constructor(props: Props) {
+    super(props);
+  }
+  
+  componentDidUpdate(){
+    if(!this.props.isFetching){
+      if(this.props.closeAction){
+        this.props.closeAction();
+      }
+      this.props.closeModal();
+    }
```

Comment:
Why does this spinner modal do actions passed in by parent components?

I think it makes sense for a spinner to be UI only, the parent component should handle side effects.

Reply:
This was my idea. Do you have ideas for how to handle this logic in the parent component?

@nnmkhang maybe we do something like `prevProps.isFetching === true && this.props.isFetching === false` inside of `componentDidUpdate()` ? I think that might work, you can try it

Reply:
Kody do you know a way to trigger the call on the falling edge of isFetching? that is what we are trying to do